src/autocomplete/index.ts

```diff
@@ -95,14 +97,17 @@ class Autocomplete {
     if (activate) {
       this.state.selected = index;
       if (indexExists) {
-        this.updateProducts(target);
+        this.updateProducts(target, updateQuery);
       }
     }
   }
 
-  updateProducts({ dataset: { query: selectedQuery, refinement, field, pastPurchase } }: HTMLElement) {
+  // tslint:disable-next-line:max-line-length
+  updateProducts({ dataset: { query: selectedQuery, refinement, field, pastPurchase } }: HTMLElement, updateQuery: boolean = true) {
```

Comment:
Also, setting a default `true` for updateQuery is no use here since you use double negation for `this.config.autocomple.hoverAutoFill`. If there is no set value in config, it will be always false and not take the default true

Reply:
If there is no value set in config then it will be `undefined`, which means it will then take the default value `true`.

Reply:
Also `const updateQuery = !!this.config.autocomplete.hoverAutoFill;` is needed here because we still want to update the query when the user presses the up and down keys. (and also why the default values for updateQuery is true for setActivation and updateProducts)